oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/DefaultOAuth2AuthorizationRequestResolverTests.java

```diff
@@ -226,4 +226,37 @@ public void resolveWhenAuthorizationRequestHasActionParameterLoginThenRedirectUr
 		OAuth2AuthorizationRequest authorizationRequest = this.resolver.resolve(request);
 		assertThat(authorizationRequest.getAuthorizationRequestUri()).matches("https://example.com/login/oauth/authorize\\?response_type=code&client_id=client-id-2&scope=read%3Auser&state=.{15,}&redirect_uri=http%3A%2F%2Flocalhost%2Flogin%2Foauth2%2Fcode%2Fregistration-id-2");
 	}
+
+	//gh-5760
+	@Test
+	public void resolveWhenAuthorizationUriHasQueryParametersThenAuthorizationURIIncludesAdditionalQueryParameters() {
+		String queryParams = "queryparam=test&queryparam2=test&queryparam3=a test with spaces";
+		ClientRegistration clientRegistration = this.registration2;
+		String requestUri = this.authorizationRequestBaseUri + "/" + clientRegistration.getRegistrationId()+"?"+queryParams;
+
+		MockHttpServletRequest request = new MockHttpServletRequest("GET", requestUri);
+		request.setServletPath(requestUri);
```

Comment:
I think this is part of the reason the manual parsing is necessary. The Javadoc of servletPath states:

>  but does not include any extra path information or a query string

This means that the value that is being set here is invalid. It should not include the query string.

Reply:
Thank you for this, you're right it's an invalid URL. I will try to fix this as soon as possible and resubmit the PR